public/components/common/modals/model_version_deployment_confirm_modal.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiButton,
+  EuiConfirmModal,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { Link, generatePath, useHistory } from 'react-router-dom';
+
+import { useOpenSearchDashboards } from '../../../../../../src/plugins/opensearch_dashboards_react/public';
+import { mountReactNode } from '../../../../../../src/core/public/utils';
+import { routerPaths } from '../../../../common';
+import { APIProvider } from '../../../apis/api_provider';
+
+import { ModelVersionErrorDetailsModal } from './model_version_error_details_modal';
+
+export const ModelVersionDeploymentConfirmModal = ({
+  id,
+  mode,
+  name,
+  version,
+  closeModal,
+}: {
+  id: string;
+  mode: 'deploy' | 'undeploy';
+  name: string;
+  version: string;
+  closeModal: () => void;
+}) => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const {
+    services: { notifications, overlays },
+  } = useOpenSearchDashboards();
+  const isDeployMode = mode === 'deploy';
+  const history = useHistory();
+
+  const handleConfirm = useCallback(async () => {
+    setIsSubmitting(true);
+    let result;
+    const modelVersionUrl = history.createHref({
+      pathname: generatePath(routerPaths.modelVersion, { id }),
```

Comment:
Q: what's the difference between `history.createHref` and `generatePath`?

Reply:
history.createHref will include the basename part. We can't assign a generatePath string to any non react-router Link component.